controllers/clusteroperator_controller_test.go

```diff
@@ -0,0 +1,182 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	timeout = time.Second * 10
+)
+
+var _ = Describe("Cluster Operator status controller", func() {
+	var operator *configv1.ClusterOperator
+	var operatorController *CloudOperatorReconciler
+
+	BeforeEach(func() {
+		operatorController = &CloudOperatorReconciler{
+			Client:           cl,
+			Scheme:           scheme.Scheme,
+			ManagedNamespace: defaultManagementNamespace,
+		}
+		operator = &configv1.ClusterOperator{}
+		operator.SetName(clusterOperatorName)
+	})
+
+	AfterEach(func() {
+		os.Unsetenv(releaseVersionEnvVariableName)
+		err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{})
+		if err == nil || !apierrors.IsNotFound(err) {
+			Expect(cl.Delete(context.Background(), operator)).To(Succeed())
+		}
+		Expect(apierrors.IsNotFound(cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{}))).To(BeTrue())
+	})
+
+	type testCase struct {
+		releaseVersionEnvVariableValue string
+		namespace                      string
+		existingCO                     *configv1.ClusterOperator
+	}
+
+	DescribeTable("should ensure Cluster Operator status is present",
+		func(tc testCase) {
+			By("Running test case")
```

Comment:
By statements are more useful when they describe parts of the test, so for example you can do:
```
By("Setting the release version", func() {
  expectedVersion := unknownVersionValue
  if tc.releaseVersionEnvVariableValue != "" {
    expectedVersion = tc.releaseVersionEnvVariableValue
    Expect(os.Setenv(releaseVersionEnvVariableName, tc.releaseVersionEnvVariableValue)).To(Succeed())
  }
})
```
Currently this By doesn't add much value